### DISCUSSION TITLE
Using edge PrismaClient feature on Nestjs

### DISCUSSION THREAD

**Discussion by mohrazzak** [prisma/prisma#21204] [Q&A] [answered]
### Question

Hello great devs, I am encountring with an issue,
When declaring my PrismaService I cant get the type of the edge PrismaClient feature
lets make that more clear
```ts
import { PrismaClient, Prisma } from '@prisma/client/edge';
import { withAccelerate } from '@prisma/extension-accelerate';
//etc...

@Injectable()
export class PrismaService extends PrismaClient implements OnModuleInit {
  constructor(
  ) {
    super({
      datasources: {
        db: {
          url: 'my db url which starts with prisma and has api_key'
        },
      },
    });
  }
  async onModuleInit() {
    await this.$connect();
  }
}
```
And now when using this on a controller it gives a `TS` error which means that cacheStrategy is not in the prismaService

```ts
  async findAll(email: string): Promise<StoreOwnerEntity | null> {
    return this.prismaService.storeOwner.findMany({
      cacheStrategy: { swr: 60, ttl: 60 }, // error TS
    });
  }
  ```
  
  However if i did this (which is not nestjs prefeered way and non standard way in nestjs)
  ```ts
  async findAll(email: string): Promise<StoreOwnerEntity | null> {
      const prisma = new PrismaClient().$extends(withAccelerate());
    return await prisma.storeOwner.findMany({
      cacheStrategy: { swr: 60, ttl: 60 }, // No error appears
    });
  }
  ```
  
  My question is how to assign the `  const prisma = new PrismaClient().$extends(withAccelerate());` type to the PrismaService i am initializing in my Nestjs App

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @mohrazzak 👋 

Thank you for raising this question.

> My question is how to assign the   `const prisma = new PrismaClient().$extends(withAccelerate());` type to the PrismaService i am initializing in my Nestjs App

Adding a client extension to the PrismaService class in Nestjs does not mutate it. For example, this code **does not help**:
```ts
export class PrismaService extends PrismaClient implements OnModuleInit {
  constructor(
  ) {
    super({
      datasources: {
        db: {
          url: ""
        },
      },
    });
    this.$extends(withAccelerate()); // this line does nothing.
  }
  async onModuleInit() {
    await this.$connect();
  }
}
```

In your controller, you can do this instead:

```ts
import { withAccelerate } from '@prisma/extension-accelerate';

  async findAll(email: string): Promise<StoreOwnerEntity | null> {
    return this.prismaService.$extends(withAccelerate()).storeOwner.findMany({
      cacheStrategy: { swr: 60, ttl: 60 }, 
    });
  }
```

This should clear the error you had with Typescript.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Kareszrk** [reply]
Thank you, your code works!
